File: original_recipe/bbcursive/src/main/java/bbcursive/lib/infix_.java
package bbcursive.lib;

import bbcursive.ann.Infix;
import bbcursive.std;
import org.jetbrains.annotations.NotNull;

import java.nio.ByteBuffer;
import java.util.Arrays;
import java.util.function.UnaryOperator;

public interface infix_ {
    @NotNull
    @Infix
static UnaryOperator<ByteBuffer> infix(UnaryOperator<ByteBuffer>... allOf) {
    return new ByteBufferUnaryOperator(allOf);

}

    @Infix
class ByteBufferUnaryOperator implements UnaryOperator<ByteBuffer> {
        private final UnaryOperator<ByteBuffer>[] allOf;

        public ByteBufferUnaryOperator(UnaryOperator<ByteBuffer>... allOf) {
            this.allOf = allOf;
        }

        @NotNull
        @Override
        public String toString() {
            return "infix"+ Arrays.deepToString(allOf);
        }

        @Override
        public ByteBuffer apply(ByteBuffer buffer) {

            return std.bb(buffer, allOf);
        }
    }
}

File: original_recipe/bbcursive/src/main/java/bbcursive/lib/infix_.kt
package bbcursive.lib

import bbcursive.ann.Infix
import bbcursive.func.UnaryOperator
import bbcursive.std
import java.nio.ByteBuffer
import java.util.*

interface infix_ {
    @Infix

    class infix(
        vararg val allOf: UnaryOperator<ByteBuffer>
    ) : UnaryOperator<ByteBuffer>
 {

        override operator fun invoke(p1: ByteBuffer): ByteBuffer {
            return std.bb(if (p1 != std.NULL_BUFF) p1 else std.NULL_BUFF, *allOf) as ByteBuffer
        }
        override fun toString(): String {
            return "infix" + Arrays.deepToString(allOf)
        }
    }

    companion object {
        @Infix
        fun infix(
            vararg allOf: UnaryOperator<ByteBuffer>
        ): UnaryOperator<ByteBuffer> {
            return infix(*allOf)
        }
    }
}